modules/msg/src/main/MsgApi.scala

```diff
@@ -329,6 +330,47 @@ final class MsgApi(
           date <- msg.getAsOpt[Instant]("date")
         yield (text, date)).toList
 
+  // include responses from the other user
+  // String is the thread id
+  def modFullCommsExport(userId: UserId): Source[(String, NonEmptyList[Msg]), ?] =
+    colls.thread
+      .aggregateWith[Bdoc](readPreference = ReadPref.priTemp): framework =>
+        import framework.*
+        List(
+          Match($doc("users" -> userId)),
+          Sort(Descending("lastMsg.date")),
+          Project($id(true)),
+          PipelineOperator:
+            $lookup.pipelineFull(
+              from = colls.msg.name,
+              as = "msgs",
+              let = $doc("t" -> "$_id"),
+              pipe = List(
+                $doc(
+                  "$match" ->
+                    $expr:
+                      $and(
+                        $doc("$eq" -> $arr("$tid", "$$t")),
+                        $doc:
+                          "$not" -> $doc:
+                            "$regexMatch" -> $doc(
+                              "input" -> "$text",
+                              "regex" -> "You received this because you are (subscribed to messages|part) of the team"
```

Comment:
Quite similar to `recentByForMod`, and `allMessagesOf` but different enough to be hard to refactor, especially if keeping the streaming is required.